packages/salesforcedx-vscode-apex/src/commands/apexActionController.ts

```diff
@@ -390,4 +447,34 @@ export class ApexActionController {
 
     return operations;
   };
+
+  /**
+   * Reads sfdx-project.json and checks if decomposeExternalServiceRegistrationBeta is enabled.
+   * @returns boolean - true if sfdx-project.json contains decomposeExternalServiceRegistrationBeta
+   */
+  private isESRDecomposed = async (): Promise<boolean> => {
```

Comment:
I see you call this in multiple places. Can you call it just once in createApexAction and then just pass it along?